### DISCUSSION TITLE
Wrong library path for newly compiled OpenSSL?

### DISCUSSION THREAD

**Discussion by 0xdeadbeer** [openssl/openssl#23405] [Q&A] [answered]
Whenever I compile OpenSSL (the libs + the tool) and want to execute the tool (without installing it), I get the following error: 
```bash
./apps/openssl: /usr/lib/libcrypto.so.3: version `OPENSSL_3.3.0' not found (required by ./apps/openssl)
```

Is there any way I can tell OpenSSL to not dig shared libraries from where they'd usually be if I were to install them? And rather get it to use the ones I have just built? 

I gave it another run with the following config: `./config --prefix=$(pwd) --openssldir=$(pwd) -d` but that resulted in an identical error - with the same identical `/usr/lib/` path that it was trying to load libs from

Any ideas?

**Comment by 0xdeadbeer** [comment, marked as answer]
Solved it by setting the `LD_LIBRARY_PATH` env variable to the OpenSSL project's root directory before executing the binary
```bash
export LD_LIBRARY_PATH=$(pwd)
./apps/openssl
```

**Reply by mattcaswell** [reply]
Note that there is some discussion about this issue and options for resolving it in the section "Shared libraries and installation in non-default locations" in the NOTES-UNIX.md file:

https://github.com/openssl/openssl/blob/master/NOTES-UNIX.md
